Cauldron/GameEventParser.cs

```diff
@@ -106,18 +112,37 @@ private bool IsNextHalfInning(Game oldState, Game newState)
 					(newState.inning - oldState.inning <= 1));
 		}
 
-		private bool IsStartOfNextAtBat(Game oldState, Game newState)
+		private bool IsStartOfInningMessage(Game newState) {
+			return (newState.lastUpdate.Contains("Top of ") || newState.lastUpdate.Contains("Bottom of "));
+		}
+
+		private bool IsGameNearlyOver(Game oldState, Game newState)
+		{
+			return ((newState.inning >= 8) &&
+					(oldState.halfInningOuts == 2 && newState.halfInningOuts == 0) &&
+					((!newState.topOfInning) || (newState.topOfInning && newState.homeScore > newState.awayScore)));
+		}
+
+		private bool IsStartOfNextAtBat(Game oldState, Game newState, int tolerance = 0)
```

Comment:
I've realized we're using two different paradigms here :) My functions that check the old state always just use m_oldState directly instead of passing it in. Dunno if it matters or which is more "correct", just interesting